src/main.rs

```diff
@@ -50,10 +50,48 @@ ow=0:or=0;38;5;16;48;5;203:no=0:ex=1;38;5;203:cd=0;38;5;203;48;5;236:mi=0;38;5;1
 ";
 
+fn main() {
+    let result = run();
+    match result {
+        Ok(exit_code) => {
+            process::exit(exit_code.into());
+        }
+        Err(err) => {
+            eprintln!("[fd error]: {:#}", err);
+            process::exit(ExitCode::GeneralError.into());
+        }
+    }
+}
+
 fn run() -> Result<ExitCode> {
     let matches = app::build_app().get_matches_from(env::args_os());
 
-    // Set the current working directory of the process
+    set_working_dir(&matches)?;
+    let current_directory = Path::new(".");
+    ensure_current_directory_exists(&current_directory)?;
+    let search_paths = extract_search_paths(&matches, current_directory)?;
+
+    let pattern = extract_search_pattern(&matches)?;
+    check_pattern_is_not_path(&matches, pattern)?;
+    let pattern_regex = build_pattern_regex(&matches, pattern)?;
+
+    // The search will be case-sensitive if the command line flag is set or
+    // if the pattern has an uppercase character (smart case).
+    let case_sensitive = !matches.is_present("ignore-case")
+        && (matches.is_present("case-sensitive") || pattern_has_uppercase_char(&pattern_regex));
+
+    let path_separator = matches
+        .value_of("path-separator")
+        .map_or_else(filesystem::default_path_separator, |s| Some(s.to_owned()));
+    check_path_separator_length(path_separator.as_deref())?;
+
+    let config = construct_options(matches, case_sensitive, path_separator)?;
```

Comment:
I know the problem was there before, but we should really use a consistent naming: config or options.